### PR TITLE
use sun2000 defaults, remove kW unit and only use W

### PR DESCRIPTION
**Pull request by daniel309**

### CODE REVIEW

File: sun2000_modbus/registers.py [hunk 1]
```diff
@@ -138,17 +138,17 @@ class BatteryEquipmentRegister(Enum):
     GridChargeCutoffSOC = Register(47088, 1, datatypes.DataType.UINT16_BE, 10, "%", AccessType.RW, None)
     # ForcibleChargeDischarge = Register(47100, 1, datatypes.DataType.UINT16_BE, 1, None, AccessType.WO, mappings.ForcibleChargeDischarge) # disabled because not readable (AccessType.WO)
     FixedChargingAndDischargingPeriods = Register(47200, 41, datatypes.DataType.MULTIDATA, None, None, AccessType.RW, None)
-    PowerOfChargeFromGrid = Register(47242, 2, datatypes.DataType.INT32_BE, 100, "kW", AccessType.RW, None)
-    MaximumPowerOfChargeFromGrid = Register(47244, 2, datatypes.DataType.INT32_BE, 100, "kW", AccessType.RW, None)
+    PowerOfChargeFromGrid = Register(47242, 2, datatypes.DataType.INT32_BE, 1, "W", AccessType.RW, None)
+    MaximumPowerOfChargeFromGrid = Register(47244, 2, datatypes.DataType.INT32_BE, 1, "W", AccessType.RW, None)
```
Comment (olivergregorius) [inline sun2000_modbus/registers.py:142]: Similar to https://github.com/olivergregorius/sun2000_modbus/pull/18/files#r1067455325
Gain must be adjusted to be 0.1, data type and all values must be changed accordingly.

File: sun2000_modbus/registers.py [hunk 2]
```diff
@@ -138,17 +138,17 @@ class BatteryEquipmentRegister(Enum):
...
     ForcibleChargeDischargeSettingMode = Register(47246, 1, datatypes.DataType.UINT16_BE, 1, None, AccessType.RW, mappings.ForcibleChargeDischargeSettingMode)
-    ForcibleChargePower = Register(47247, 2, datatypes.DataType.INT32_BE, 100, "kW", AccessType.RW, None)
-    ForcibleDischargePower = Register(47249, 2, datatypes.DataType.INT32_BE, 100, "kW", AccessType.RW, None)
+    ForcibleChargePower = Register(47247, 2, datatypes.DataType.INT32_BE, 1, "W", AccessType.RW, None)
+    ForcibleDischargePower = Register(47249, 2, datatypes.DataType.INT32_BE, 1, "W", AccessType.RW, None)
```
Comment (olivergregorius) [inline sun2000_modbus/registers.py:145]: See https://github.com/olivergregorius/sun2000_modbus/pull/18/files#r1067457689

File: sun2000_modbus/registers.py [hunk 3]
```diff
@@ -138,17 +138,17 @@ class BatteryEquipmentRegister(Enum):
...
     TimeOfUseChargingAndDischargingPeriods = Register(47255, 43, datatypes.DataType.MULTIDATA, None, None, AccessType.RW, None)
     ExcessPVEnergyUseInTOU = Register(47299, 1, datatypes.DataType.UINT16_BE, 1, None, AccessType.RW, mappings.ExcessPVEnergyUseInTOU)
     ActivePowerControlMode = Register(47415, 1, datatypes.DataType.UINT16_BE, 1, None, AccessType.RW, mappings.ActivePowerControlMode)
     MaximumFeedGridPowerInKW = Register(47416, 2, datatypes.DataType.INT32_BE, 1000, "kW", AccessType.RW, None)
     MaximumFeedGridPowerInPercentage = Register(47418, 1, datatypes.DataType.INT16_BE, 10, "%", AccessType.RW, None)
-    MaximumChargeFromGridPower = Register(47590, 2, datatypes.DataType.INT32_BE, 100, "kW", AccessType.RW, None)
+    MaximumChargeFromGridPower = Register(47590, 2, datatypes.DataType.INT32_BE, 1, "W", AccessType.RW, None)
```
Comment (olivergregorius) [inline sun2000_modbus/registers.py:151]: See https://github.com/olivergregorius/sun2000_modbus/pull/18/files#r1067457689

File: sun2000_modbus/registers.py [hunk 4]
```diff
@@ -199,7 +199,7 @@ class BatteryEquipmentRegister(Enum):
     Unit1BatteryPack1Voltage = Register(38235, 1, datatypes.DataType.UINT16_BE, 10, "V", AccessType.RO, None)
     Unit1BatteryPack1Current = Register(38236, 1, datatypes.DataType.INT16_BE, 10, "A", AccessType.RO, None)
     Unit1BatteryPack1SOC = Register(38229, 1, datatypes.DataType.UINT16_BE, 10, "%", AccessType.RO, None)
-    Unit1BatteryPack1ChargeDischargePower = Register(38233, 2, datatypes.DataType.INT32_BE, 100, "kW", AccessType.RO, None)
+    Unit1BatteryPack1ChargeDischargePower = Register(38233, 2, datatypes.DataType.INT32_BE, 1, "W", AccessType.RO, None)
```
Comment (olivergregorius) [inline sun2000_modbus/registers.py:202]: See https://github.com/olivergregorius/sun2000_modbus/pull/18/files#r1067457689

File: sun2000_modbus/registers.py [hunk 5]
```diff
@@ -213,7 +213,7 @@ class BatteryEquipmentRegister(Enum):
     Unit1BatteryPack2Voltage = Register(38277, 1, datatypes.DataType.UINT16_BE, 10, "V", AccessType.RO, None)
     Unit1BatteryPack2Current = Register(38278, 1, datatypes.DataType.INT16_BE, 10, "A", AccessType.RO, None)
     Unit1BatteryPack2SOC = Register(38271, 1, datatypes.DataType.UINT16_BE, 10, "%", AccessType.RO, None)
-    Unit1BatteryPack2ChargeDischargePower = Register(38275, 2, datatypes.DataType.INT32_BE, 100, "kW", AccessType.RO, None)
+    Unit1BatteryPack2ChargeDischargePower = Register(38275, 2, datatypes.DataType.INT32_BE, 1, "W", AccessType.RO, None)
```
Comment (olivergregorius) [inline sun2000_modbus/registers.py:216]: See https://github.com/olivergregorius/sun2000_modbus/pull/18/files#r1067457689

File: sun2000_modbus/registers.py [hunk 6]
```diff
@@ -227,7 +227,7 @@ class BatteryEquipmentRegister(Enum):
     Unit1BatteryPack3Voltage = Register(38319, 1, datatypes.DataType.UINT16_BE, 10, "V", AccessType.RO, None)
     Unit1BatteryPack3Current = Register(38320, 1, datatypes.DataType.INT16_BE, 10, "A", AccessType.RO, None)
     Unit1BatteryPack3SOC = Register(38313, 1, datatypes.DataType.UINT16_BE, 10, "%", AccessType.RO, None)
-    Unit1BatteryPack3ChargeDischargeStatus = Register(38317, 2, datatypes.DataType.INT32_BE, 100, "kW", AccessType.RO, None)
+    Unit1BatteryPack3ChargeDischargeStatus = Register(38317, 2, datatypes.DataType.INT32_BE, 1, "W", AccessType.RO, None)
```
Comment (olivergregorius) [inline sun2000_modbus/registers.py:230]: See https://github.com/olivergregorius/sun2000_modbus/pull/18/files#r1067457689

File: sun2000_modbus/registers.py [hunk 7]
```diff
@@ -241,7 +241,7 @@ class BatteryEquipmentRegister(Enum):
     Unit2BatteryPack1Voltage = Register(38361, 1, datatypes.DataType.UINT16_BE, 10, "V", AccessType.RO, None)
     Unit2BatteryPack1Current = Register(38362, 1, datatypes.DataType.INT16_BE, 10, "A", AccessType.RO, None)
     Unit2BatteryPack1SOC = Register(38355, 1, datatypes.DataType.UINT16_BE, 10, "%", AccessType.RO, None)
-    Unit2BatteryPack1ChargeDischargePower = Register(38359, 2, datatypes.DataType.INT32_BE, 100, "kW", AccessType.RO, None)
+    Unit2BatteryPack1ChargeDischargePower = Register(38359, 2, datatypes.DataType.INT32_BE, 1, "W", AccessType.RO, None)
```
Comment (olivergregorius) [inline sun2000_modbus/registers.py:244]: See https://github.com/olivergregorius/sun2000_modbus/pull/18/files#r1067457689

File: sun2000_modbus/registers.py [hunk 8]
```diff
@@ -255,7 +255,7 @@ class BatteryEquipmentRegister(Enum):
     Unit2BatteryPack2Voltage = Register(38403, 1, datatypes.DataType.UINT16_BE, 10, "V", AccessType.RO, None)
     Unit2BatteryPack2Current = Register(38404, 1, datatypes.DataType.INT16_BE, 10, "A", AccessType.RO, None)
     Unit2BatteryPack2SOC = Register(38397, 1, datatypes.DataType.UINT16_BE, 10, "%", AccessType.RO, None)
-    Unit2BatteryPack2ChargeDischargePower = Register(38401, 2, datatypes.DataType.INT32_BE, 100, "kW", AccessType.RO, None)
+    Unit2BatteryPack2ChargeDischargePower = Register(38401, 2, datatypes.DataType.INT32_BE, 1, "W", AccessType.RO, None)
```
Comment (olivergregorius) [inline sun2000_modbus/registers.py:258]: See https://github.com/olivergregorius/sun2000_modbus/pull/18/files#r1067457689

File: sun2000_modbus/registers.py [hunk 9]
```diff
@@ -269,7 +269,7 @@ class BatteryEquipmentRegister(Enum):
     Unit2BatteryPack3Voltage = Register(38445, 1, datatypes.DataType.UINT16_BE, 10, "V", AccessType.RO, None)
     Unit2BatteryPack3Current = Register(38446, 1, datatypes.DataType.INT16_BE, 10, "A", AccessType.RO, None)
     Unit2BatteryPack3SOC = Register(38439, 1, datatypes.DataType.UINT16_BE, 10, "%", AccessType.RO, None)
-    Unit2BatteryPack3ChargeDischargePower = Register(38443, 2, datatypes.DataType.INT32_BE, 100, "kW", AccessType.RO, None)
+    Unit2BatteryPack3ChargeDischargePower = Register(38443, 2, datatypes.DataType.INT32_BE, 1, "W", AccessType.RO, None)
```
Comment (olivergregorius) [inline sun2000_modbus/registers.py:272]: See https://github.com/olivergregorius/sun2000_modbus/pull/18/files#r1067457689

**Review by olivergregorius — CHANGES_REQUESTED** [review]
Thanks for your contribution!

File: sun2000_modbus/inverter.py [hunk 10]
```diff
@@ -10,7 +10,7 @@
 
 
 class Sun2000:
-    def __init__(self, host, port=502, timeout=5, wait=2, unit=0):
+    def __init__(self, host, port=502, timeout=5, wait=2, unit=1):
```
[the remark this answers is not in the document]
Comment (daniel309) [reply sun2000_modbus/inverter.py:13]: interesting. I did google around a bit and it seems to vary by model. let me put a comment and revert back to 0

**Comment by daniel309** [comment]
Thanks, my intention was to standardize all registers that report power and dont have a unit in their name in W. Current master has a mix (e.g. Inverter InputPower and ActivePower are kW, vs. Battery ChargeDischargePower and Meter ActivePower are W already). I found this quite annoying and that caused me to start this PR. 

Btw. I messed up the PR. Wanted to merge into my fork's master and only later open a PR for upstream master.

Let me fix my stuff and then get back. I also wanted to change some of the isConnected logic and add an option to always close the tcp connection. My inverter is very sensitive to leaving the tcp connection open, others reported similar problems

https://www.elektroda.com/rtvforum/topic3828425-30.html

https://skyboo.net/2022/02/huawei-sun2000-why-using-a-usb-dongle-for-monitoring-is-not-a-good-idea/

let me fix all that in my fork, fix tests and then open a PR for when I am done. Will get back. Closing this PR for now.

**Pull request closed without merging**